### ISSUE TITLE
ENOTFOUND - allow overriding of advertised host

### ISSUE THREAD

**Issue by cdaringe** [SOHU-Co/kafka-node#1052]
# problem

- i cannot read my kafka data because the client tries to access the hostname that kafka provides it

# discussion

- i have my kafka instance in a private cluster, which has its own DNS
- during development, i periodically port forward or ssh onto that machine to inspect kafka

i try to use use `kafka-node` providing a `kafkhaHost: <IP>`, however, on resolving of the `kafkaHost`, the client insists on using the kafka defined hostname.  because that environment (in docker) uses its own internal DNS, of course the _new_ broker configs all point to a hostname that is unaccessible to me.

i want to be able to tell the client, "please just keep using the `kafkaHost` i provided you--do not trust the advertised host that kafka reports out, because that hostname is on a completly different network"

**Comment by hyperlink** [comment]
could you update your `/etc/hosts` file to get the behavior you want?

**Comment by cdaringe** [comment]
yes, that's the only solution ATM.  However, this is a system wide side effect that I've already forgotten to undo.  further, my integration tests want to point to various instances that are dynamically torn up and down, which makes the meddling of etc hosts a somewhat difficult proposition (e.g. don't open up perms on etc hosts just for a test run).

thanks for the response :)
